augur/utils.py

```diff
@@ -675,19 +691,56 @@ def read_bed_file(bed_file):
     list of int:
         Sorted list of unique zero-indexed sites
     """
-    mask_sites = []
-    try:
-        bed = pd.read_csv(bed_file, sep='\t', header=None, usecols=[1,2],
-                          dtype={1:int,2:int}, **PANDAS_READ_CSV_OPTIONS)
-    except ValueError:
-        # Check if we have a header row. Otherwise, just fail.
-        bed = pd.read_csv(bed_file, sep='\t', header=None, usecols=[1,2],
-                          dtype={1:int,2:int}, skiprows=1, **PANDAS_READ_CSV_OPTIONS)
-        print("Skipped row 1 of %s, assuming it is a header." % bed_file)
-    for _, row in bed.iterrows():
-        mask_sites.extend(range(row[1], row[2]))
+    in_header = True
+    initial_chrom_value = ""
+    mask_sites: list[int] = []
+
+    bed_file_size = os.path.getsize(bed_file)
+
+    # Only try to read the file if it is larger than zero bytes;
+    # otherwise return an empty list
+    if bed_file_size == 0:
+        return mask_sites
+
+    with open(bed_file, 'r', newline='') as bed_fh:
+        for line in bed_fh:
+            row = line.split("\t")
+
+            # skip all lines starting with "#" regardless of where they are
+            if row[0].startswith("#"):
+                continue
+            elif row[0].lower().startswith(("browser", "chrom", "track")):
+                if in_header:
+                    # ignore header lines
+                    continue
+                else:
+                    # once a data line has been seen, header lines are not allowed
+                    raise AugurError(
+                        f"BED file {bed_file} has a header line after data lines."+
+                        f"\nInvalid header:\n{repr(row)}"
+                    )
+
+            # if we get to here, this is a data line, no going back now
+            in_header = False
+
+            # extract the values we care about
+            chrom, start, end, *_ = row
```

Comment:
This'll be an uncaught `ValueError` if the row doesn't have at least 3 (tab-separated) fields. Not sure whether it's preferable to raise `AugurError` or just drop the line and continue... I'd probably go with the former?